src/MongoDBRepository.php

```diff
@@ -39,19 +39,19 @@ public function __construct(Collection $collection, Serializer $serializer, stri
     /**
      * {@inheritdoc}
      */
-    public function save(Identifiable $model)
+    public function save(Identifiable $model): void
     {
         Assert::isInstanceOf($model, $this->class);
 
         $normalized = $this->normalizeIdentifiable($model);
 
-        $this->collection->insertOne($normalized);
+        $this->collection->replaceOne(['_id' => $model->getId()], $normalized, ['upsert' => true]);
```

Comment:
I guessed that the idea was to have only 1 method that can be used to save a freshly new ReadModel but also to update an existing ReadModel. That's why it seems to insertOne() should be converted to replaceOne() with upsert flag